.github/workflows/release.yaml

```diff
@@ -0,0 +1,498 @@
+            ${{ steps.go-env.outputs.cache }}
+            ${{ steps.go-env.outputs.modcache }}
+          key: push-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: push-go-
+      - name: Set up Git config
+        run: |
+          git config user.name actions-user
+          git config user.email actions@github.com
+      - name: Fetch tags
+        env:
+          version: ${{ needs.prepare1.outputs.version }}
+        run: |
+          git fetch origin \
+            tag "${version}" \
+            tag engine/badgerengine/"${version}" \
+            tag cmd/genji/"${version}"
+      - name: Merge release
+        env:
+          version: ${{ needs.prepare1.outputs.version }}
+        run: |
+          echo "Before merge: $(git rev-parse HEAD)"
+          git merge \
+            refs/tags/"${version}" \
+            refs/tags/engine/badgerengine/"${version}" \
+            refs/tags/cmd/genji/"${version}"
+          echo "After merge: $(git rev-parse HEAD)"
+      - name: Add replace directives
+        run: |
+          go mod tidy
+          git diff
+          git add go.mod go.sum
+          (
+            cd engine/badgerengine
+            go mod edit -replace=github.com/genjidb/genji=../../ 
+            go mod tidy
+            git diff
+            git add go.mod go.sum
+          )
+          (
+            cd cmd/genji
+            go mod edit -replace=github.com/genjidb/genji=../../
+            go mod edit -replace=github.com/genjidb/genji/engine/badgerengine=../../engine/badgerengine/
+            go mod tidy
+            git diff
+            git add go.mod go.sum
+          )
+          git commit --allow-empty -m 'Add replace directives [skip ci]'
+      - name: Push PR branch
+        id: branch
+        env:
+          version: ${{ needs.prepare1.outputs.version }}
+        run: |
+          branch=merge-release-"${version}"
+          git checkout -b "${branch}"
+          git push origin "${branch}"
+          echo "::set-output name=name::${branch}"
+
+  merge:
+    name: Create PR
+    runs-on: ubuntu-latest
+    needs:
+      - prepare1
+      - push
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2.3.4
+      - name: Create PR
+        run: gh pr create -R="${repo}" -t="${title}" -b="${body}" -B="${base}" -H="${head}"
+        env:
+          repo: ${{ github.repository }}
+          base: ${{ needs.prepare1.outputs.branch }}
+          head: ${{ needs.push.outputs.branch }}
+          # Ensure that default merge commit body contains [skip ci] so that we don’t trigger release workflow.
+          # See also https://github.blog/changelog/2021-02-08-github-actions-skip
+          title: 'Merge release tags for ${{ needs.prepare1.outputs.version }} [skip ci]'
+          body: |
+            This PR merges ${{ needs.prepare1.outputs.version }} release tags.
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  # Remove tags and draft release on failure.
+  #
+  rollback:
```

Comment:
We should also set concurrency group for release workflow. See https://github.blog/changelog/2021-04-19-github-actions-limit-workflow-run-or-job-concurrency/